lib/signal-manager.js

```diff
@@ -1,17 +1,19 @@
 const runningProcs = new Set()
 let handlersInstalled = false
 
+// NOTE: these signals aren't actually forwarded anywhere. they're trapped and
+// ignored until all child processes have exited. in our next breaking change
+// we should rename this
 const forwardedSignals = [
   'SIGINT',
   'SIGTERM',
 ]
 
-const handleSignal = signal => {
-  for (const proc of runningProcs) {
-    proc.kill(signal)
-  }
-}
```

Comment:
I think this assumes npm is the process group leader and that signals to npm's pid therefore get propagated to all other pids in the group.

That's a bad assumption though. It's a breaking - and, I suspect, unintentional - change for processes that switch process groups. Those stay behind now.

Reply:
I think the author might have confused terminating the program with "CTRL + C" (which still works) with sending a SIGTERM signal, for example, by using kill -SIGTERM $PID. This particular functionality seems to have been broken by this commit.